### PR TITLE
Akshay Changes Alan Weekly Summaries

### PR DESCRIPTION
**Pull request by agujjar2206**
# Description
This PR is a replacement for Alan's PR #593 and includes all the changes from that PR, as well as a few minor updates, such as removing unused comments.

## Main changes explained:
- Implement the functionality from Alan's PR #593
- Remove unused comments from various files in the codebase

## How to test:
- Check out the current branch
- Run `npm install` to ensure all dependencies are installed
- Start the application and verify that it runs smoothly and that the new functionality from PR #593 is working correctly

## Screenshots or videos of changes (if applicable):
None, as the changes are only related to comments and do not affect UI or functionality.

Note:
This PR includes the changes from Alan's PR #593 and additional changes related to removing unused comments. The code has been tested locally, and no issues have been identified.

### CODE REVIEW

**Comment by one-community** [comment]
NOTE: This PR replaces PR #593:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/593

File: src/components/UserProfile/UserProfile.jsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import React, { useState, useEffect, useRef } from 'react';
```
Comment (KaixiangGU) [inline src/components/UserProfile/UserProfile.jsx:1]: Still missing this line

Comment (KaixiangGU) [reply src/components/UserProfile/UserProfile.jsx:1]: Just add it back should be fine. Alan deleted it accidentally in his most recent commit.

Comment (agujjar2206) [reply src/components/UserProfile/UserProfile.jsx]: Thanks for bringing this to my attention, @KaixiangGU. I've added the missing line back to the code in my branch, so it should be included in the latest version of the PR.

Comment (KaixiangGU) [reply src/components/UserProfile/UserProfile.jsx]: Thank you! Looks good!

**Review by KaixiangGU — APPROVED** [review]
Good to go!

**Comment by one-community** [comment]
Excellent!! Thank you both!!

**Review by hetvi11110 — COMMENTED** [review]
I tested both frontend and backend PR. Both are working as mentioned. For reference, please check the comment on frontend PR.

**Pull request merged**